### ISSUE TITLE
[X86] X86SchedAlderlakeP.td _Int scalar cvt intrinsics map to different schedules to the regular instruction

### ISSUE THREAD

**Issue by RKSimon** [llvm/llvm-project#58792]
The AlderLakeP model has explicit overrides for some of the "*_Int" scalar conversion intrinsics, but not the regular def of the same instruction:
```
def ADLPWriteResGroup36 : SchedWriteRes<[ADLPPort00_01, ADLPPort05]> {
  let ResourceCycles = [1, 2];
  let Latency = 8;
  let NumMicroOps = 3;
}
def : InstRW<[ADLPWriteResGroup36, ReadDefault, ReadInt2Fpu], (instregex "^(V?)CVTSI642SSrr_Int$")>;
```
Should the _Int$ postfix be (_Int)?$ to match both cases?

**Comment by HaohaiWen** [comment]
Thanks @RKSimon for reporting that. I'll fix it.

**Comment by HaohaiWen** [comment]
VCVTSI642SSrr is CodeGenOnly opcode so schedtool won't generate valid scheduling info for it.
I'm working on a workaround to enable scheduling info for those special CodeGenOnly instructions.

**Comment by HaohaiWen** [comment]
Some CodeGenOnly but encodable X86 instructions with REX_W prefix missed In64BitMode predicates which causes assertion fails in llvm-mc. This blocks us to auto gen schedmodel. We need fix them first.

**Comment by RKSimon** [comment]
Would you mind if I manually edited the model instregex for now? This is one of a large number of scheduler model issues I'm currently trying to cleanup (I'm working on better model checks atm).

You can then regenerate later once you have fixed the underlying issues.

**Comment by HaohaiWen** [comment]
I can address instregex first (I think I can file a patch today). Then address this _Int issue.

**Comment by HaohaiWen** [comment]
Fix patch for instregex is here: https://reviews.llvm.org/D137841

**Comment by HaohaiWen** [comment]
> Would you mind if I manually edited the model instregex for now? This is one of a large number of scheduler model issues I'm currently trying to cleanup (I'm working on better model checks atm).
> 
> You can then regenerate later once you have fixed the underlying issues.

Please feel free to edit it if my patch didn't solve you problem.

**Comment by RKSimon** [comment]
Fixed by 8ce5579d7f1836d65ef6d34f351462e5b79cbda6
